pkg/kubectl/cmd/apply_test.go

```diff
@@ -216,6 +238,62 @@ func TestApplyObject(t *testing.T) {
 	}
 }
 
+// TestApplyTPR uses apply commad with ThirdPartyResources.
+func TestApplyTPR(t *testing.T) {
+	initTestErrorHandler(t)
+	nameTPR, currentTPR := readAndAnnotateTPR(t, filenameTPR)
+	pathTPR := "/thirdpartyresources/" + nameTPR
+
+	f, tf, _, ns := NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.JSONEncoder = testapi.Extensions.Codec()
+	tf.Client = &fake.RESTClient{
+		ContentConfig: restclient.ContentConfig{
+			ContentType: runtime.ContentTypeJSON,
+			GroupVersion: &unversioned.GroupVersion{
+				Group:   extensions.GroupName,
+				Version: "v1beta1",
+			},
+			NegotiatedSerializer: ns,
+		},
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == pathTPR && m == "GET":
+				bodyTPR := ioutil.NopCloser(bytes.NewReader(currentTPR))
+				return &http.Response{
+					StatusCode: 200,
+					Header:     defaultHeader(),
+					Body:       bodyTPR,
+				}, nil
+			case p == pathTPR && m == "PATCH":
```

Comment:
You are not checking if the patch has the expected field here.
Please take a look this test `func TestApplyNULLPreservation(t *testing.T)`